bib/cmd/bootc-image-builder/image.go

```diff
@@ -293,27 +195,97 @@ func manifestForISO(c *ManifestConfig, rng *rand.Rand) (*manifest.Manifest, erro
 				ImageFormat: platform.FORMAT_ISO,
 			},
 			BIOS:       true,
-			UEFIVendor: "fedora",
+			UEFIVendor: c.SourceInfo.UEFIVendor,
 		}
 	case arch.ARCH_AARCH64:
+		// aarch64 always uses UEFI, so let's enforce the vendor
+		if c.SourceInfo.UEFIVendor == "" {
+			return nil, fmt.Errorf("UEFI vendor must be set for aarch64 ISO")
+		}
 		img.Platform = &platform.Aarch64{
 			BasePlatform: platform.BasePlatform{
 				ImageFormat: platform.FORMAT_ISO,
 			},
-			UEFIVendor: "fedora",
+			UEFIVendor: c.SourceInfo.UEFIVendor,
 		}
 	}
 
 	img.OSName = "default"
 	img.Filename = "install.iso"
 
 	mf := manifest.New()
-	mf.Distro = manifest.DISTRO_FEDORA
-	runner := &runner.Fedora{Version: 39}
-	_, err := img.InstantiateManifest(&mf, c.Repos, runner, rng)
+
+	foundDistro, foundRunner, err := getDistroAndRunner(c.SourceInfo.OSRelease)
+	if err != nil {
+		return nil, fmt.Errorf("failed to infer distro and runner: %w", err)
+	}
+	mf.Distro = foundDistro
+
+	_, err = img.InstantiateManifest(&mf, nil, foundRunner, rng)
 	return &mf, err
 }
 
+func getDistroAndRunner(osRelease source.OSRelease) (manifest.Distro, runner.Runner, error) {
+	switch osRelease.ID {
+	case "fedora":
+		version, err := strconv.ParseUint(osRelease.VersionID, 10, 64)
+		if err != nil {
+			return manifest.DISTRO_NULL, nil, fmt.Errorf("cannot parse Fedora version (%s): %w", osRelease.VersionID, err)
+		}
+
+		return manifest.DISTRO_FEDORA, &runner.Fedora{
+			Version: version,
+		}, nil
+	case "centos":
+		version, err := strconv.ParseUint(osRelease.VersionID, 10, 64)
+		if err != nil {
+			return manifest.DISTRO_NULL, nil, fmt.Errorf("cannot parse CentOS version (%s): %w", osRelease.VersionID, err)
+		}
+		r := &runner.CentOS{
+			Version: version,
+		}
+		switch version {
+		case 9:
+			return manifest.DISTRO_EL9, r, nil
+		case 10:
+			return manifest.DISTRO_EL10, r, nil
+		default:
+			logrus.Warnf("Unknown CentOS version %d, using default distro for manifest generation", version)
+			return manifest.DISTRO_NULL, r, nil
+		}
+
+	case "rhel":
+		versionParts := strings.Split(osRelease.VersionID, ".")
+		if len(versionParts) != 2 {
+			return manifest.DISTRO_NULL, nil, fmt.Errorf("invalid RHEL version format: %s", osRelease.VersionID)
+		}
+		major, err := strconv.ParseUint(versionParts[0], 10, 64)
+		if err != nil {
+			return manifest.DISTRO_NULL, nil, fmt.Errorf("cannot parse RHEL major version (%s): %w", versionParts[0], err)
```

Comment:
(super nitpick) I wonder if including the full `osRelease.VersionID` from before the split gives the user more context? I am 80% confident that ParseUint() will include the string that it failed to parse already (also below)